src/vcpkg/binarycaching.cpp

```diff
@@ -782,17 +778,20 @@ namespace
                     if (fs.exists(nupkg_path, IgnoreErrors{}))
                     {
                         fs.remove(nupkg_path, VCPKG_LINE_INFO);
-                        Checks::check_exit(VCPKG_LINE_INFO,
-                                           !fs.exists(nupkg_path, IgnoreErrors{}),
-                                           "Unable to remove nupkg after restoring: %s",
-                                           nupkg_path);
+                        if (fs.exists(nupkg_path, IgnoreErrors{}))
```

Comment:
I can see an argument that this is pointless and we should remove it, after all, remove succeeded.